tests/test_diff_rigid.py

```diff
@@ -0,0 +1,216 @@
+        )
+        func_solve(
+            entities_info=rigid_solver.entities_info,
+            dofs_state=rigid_solver.dofs_state,
+            constraint_state=constraint_solver.constraint_state,
+            rigid_global_info=rigid_solver._rigid_global_info,
+            static_rigid_sim_config=rigid_solver._static_rigid_sim_config,
+            static_rigid_sim_cache_key=rigid_solver._static_rigid_sim_cache_key,
+        )
+
+    def scene_step():
+        # Mock the scene step because if we call scene.step(), it will overwrite the necessary information that we need to
+        # compute the gradients for
+        kernel_step_1(
+            links_state=rigid_solver.links_state,
+            links_info=rigid_solver.links_info,
+            joints_state=rigid_solver.joints_state,
+            joints_info=rigid_solver.joints_info,
+            dofs_state=rigid_solver.dofs_state,
+            dofs_info=rigid_solver.dofs_info,
+            geoms_state=rigid_solver.geoms_state,
+            geoms_info=rigid_solver.geoms_info,
+            entities_state=rigid_solver.entities_state,
+            entities_info=rigid_solver.entities_info,
+            rigid_global_info=rigid_solver._rigid_global_info,
+            static_rigid_sim_config=rigid_solver._static_rigid_sim_config,
+            contact_island_state=constraint_solver.contact_island.contact_island_state,
+            static_rigid_sim_cache_key=rigid_solver._static_rigid_sim_cache_key,
+        )
+        rigid_solver._func_constraint_clear()
+        constraint_solver.add_equality_constraints()
+        rigid_solver.collider.detection()
+        constraint_solver.add_frictionloss_constraints()
+        constraint_solver.add_collision_constraints()
+        constraint_solver.add_joint_limit_constraints()
+        constraint_solver_resolve()
+
+    def compute_loss(input_mass, input_jac, input_aref, input_efc_D, input_force):
+        rigid_solver._rigid_global_info.mass_mat.from_numpy(input_mass.detach().cpu().numpy())
+        constraint_solver.constraint_state.jac.from_numpy(input_jac.detach().cpu().numpy())
+        constraint_solver.constraint_state.aref.from_numpy(input_aref.detach().cpu().numpy())
+        constraint_solver.constraint_state.efc_D.from_numpy(input_efc_D.detach().cpu().numpy())
+        rigid_solver.dofs_state.force.from_numpy(input_force.detach().cpu().numpy())
+
+        # Recompute acc_smooth from the updated input variables
+        updated_acc_smooth = torch.linalg.solve(input_mass.squeeze(-1), input_force.squeeze(-1))
+        input_acc_smooth = updated_acc_smooth.unsqueeze(-1).detach().cpu().numpy()
+
+        rigid_solver.dofs_state.acc_smooth.from_numpy(input_acc_smooth)
+
+        constraint_solver_resolve()
+        output_qacc = constraint_solver.qacc.to_numpy()
+        th_output_qacc = torch.from_numpy(output_qacc).to(device=target_qacc.device)
+        loss = ((th_output_qacc - target_qacc) ** 2).mean()
+        return loss
```

Comment:
What about leveraging built-in pytest monkey patching mechanism?

We are already doing it at several places, it seems to fit perfectly for this use case. See the unit tests `test_plotter`, `test_interactive_viewer_key_press`, or `test_batch_deformable_render`. The official doc is available [here](https://docs.pytest.org/en/stable/how-to/monkeypatch.html).